### PR TITLE
Displaying city zoomed and centered after clicking on its name in Cit…

### PR DESCRIPTION
**Pull request by tomashpl**
Displaying city zoomed and centered after clicking on its name in CitiesSplit component

### CODE REVIEW

**Comment by tomashpl** [comment]
It's related to the issue: https://github.com/konradkalemba/korona.ws/issues/9

File: src/components/DataElement/CitiesSplit.jsx [hunk 1]
```diff
@@ -20,19 +23,31 @@ export default function CitiesSplit({ isLoading, data }) {
...
+          <DataContext.Consumer>{({ setPosition, setZoom }) => (
+            <ResponsiveContainer height={data.length * 32} width={'99%'}>
+              <BarChart
+                data={data}
+                layout="vertical"
+              >
+                <YAxis
+                  dataKey="city"
+                  type="category"
+                  tick={{ fill: theme.colors.contentPrimary }}
```
Comment (nowNick) [inline src/components/DataElement/CitiesSplit.jsx]: Do you think there's a way for ticks to have `cursor: pointer` ?

Comment (tomashpl) [reply src/components/DataElement/CitiesSplit.jsx]: I think yes, but to achieve this we probably should use customized tick Component. I'll try to implemet this as improvement ASAP :)

File: src/components/DataElement/CitiesSplit.jsx [hunk 2]
```diff
@@ -20,19 +23,31 @@ export default function CitiesSplit({ isLoading, data }) {
...
-            >
-              <YAxis dataKey="city" type="category" tick={{ fill: theme.colors.contentPrimary }} width={100} />
-              <XAxis type="number" hide />
-              <Tooltip
-                formatter={value => [value, 'Liczba']}
-              />
-              <Bar dataKey="count" fill={theme.colors.accent} />
-            </BarChart>
-          </ResponsiveContainer>}
+          <DataContext.Consumer>{({ setPosition, setZoom }) => (
```
Comment (konradkalemba) [inline src/components/DataElement/CitiesSplit.jsx]: Works well, but there are some improvements I would implement. First of all, in DataContext I would store just city name. Then basing on that I would find its location from `cities` array and change the map's position. I wouldn't also use `DataContext.Consumer` when you can use `useData` hook above.

File: src/components/DataElement/CitiesSplit.jsx [hunk 3]
```diff
@@ -20,19 +23,31 @@ export default function CitiesSplit({ isLoading, data }) {
...
+                data={data}
+                layout="vertical"
+              >
+                <YAxis
+                  dataKey="city"
+                  type="category"
+                  tick={{ fill: theme.colors.contentPrimary }}
+                  width={100}
+                  onClick={({ value }) => {
+                    setPosition(find(data, { city: value }).location);
```
Comment (konradkalemba) [inline src/components/DataElement/CitiesSplit.jsx]: I don't think we have to use `find` method of lodash. Doesn't `filter` method returns the same result?

**Comment by konradkalemba** [comment]
Thanks for your help. 👍 It works well but I left some comments in code.

File: src/components/DataElement/CitiesSplit.jsx (hunk 3 above)
Comment (tomashpl) [reply src/components/DataElement/CitiesSplit.jsx]: Done

File: src/components/DataElement/CitiesSplit.jsx (hunk 2 above)
Comment (tomashpl) [reply src/components/DataElement/CitiesSplit.jsx]: Done

**Comment by konradkalemba** [comment]
@tomashpl The code looks good now. :) Thanks for the changes.

**Comment by tomashpl** [comment]
Omg, that thing with cursor was that easy? Shame on me :D

File: src/components/DataElement/CitiesSplit.jsx (hunk 1 above)
Comment (konradkalemba) [reply src/components/DataElement/CitiesSplit.jsx]: Done, it was easier than expected. :D I tried adding cursor property directly to tick prop and it somehow worked.

**Comment by konradkalemba** [comment]
> Omg, that thing with cursor was that easy? Shame on me :D

Hah, I wouldn't have thought that it was that easy either. :D

**Pull request merged**